indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -175,6 +211,14 @@ public TaskConfig(
     }
     log.debug("Batch processing mode:[%s]", this.batchProcessingMode);
     this.storeEmptyColumns = storeEmptyColumns == null ? DEFAULT_STORE_EMPTY_COLUMNS : storeEmptyColumns;
+
+    if (baseTaskDirLocations == null || baseTaskDirLocations.size() == 0) {
+      File baseTaskFile = new File(defaultDir(baseTaskDir, "persistent/task"));
```

Comment:
the `baseTaskDir` configuration now is meaningful when the new `baseTaskDirLocations` is not set. For the long term evolution, should the `baseTaskDir` be marked as deprecated?